client/src/components/SelectionDialog/HistoryDatasetPicker.vue

```diff
@@ -0,0 +1,325 @@
+            selectionDialog.value?.resetPagination();
+        }
+
+        const limit = ctx.perPage;
+        const offset = (ctx.currentPage - 1) * ctx.perPage;
+        const sortDesc = ctx.sortDesc;
+        const sortBy: HistorySortByLiteral =
+            ctx.sortBy === "label" ? "name" : (ctx.sortBy as HistorySortByLiteral) || "update_time";
+        const queryDict = HistoriesFilters.getQueryDict(ctx.filter);
+
+        const { response, data, error } = await GalaxyApi().GET("/api/histories", {
+            params: {
+                query: {
+                    search: filter,
+                    view: "summary",
+                    keys: "create_time",
+                    show_own: true,
+                    offset: offset,
+                    limit: limit,
+                    q: Object.keys(queryDict),
+                    qv: Object.values(queryDict),
+                    sort_by: sortBy,
+                    sort_desc: sortDesc,
+                },
+            },
+        });
+
+        if (error) {
+            errorMessage.value = errorMessageAsString(error);
+            return [];
+        }
+
+        totalItems.value = parseInt(response.headers.get("total_matches") ?? "0");
+
+        items.value = (data as HistorySummary[]).map(historyEntryToRecord);
+
+        formatRows();
+
+        return items.value;
+    } catch (error) {
+        errorMessage.value = errorMessageAsString(error);
+        return [];
+    } finally {
+        loading.value = false;
+    }
+}
+
+async function datasetsProvider(ctx: ItemsProviderContext, selectedHistory: HistoryRecord) {
+    loading.value = true;
+
+    try {
+        const limit = ctx.perPage;
+        const offset = (ctx.currentPage - 1) * ctx.perPage;
+        const query = ctx.filter || "";
+        const querySortBy = ctx.sortBy === "time" ? "update_time" : "name";
+        const sortPrefix = ctx.sortDesc ? "-dsc" : "-asc";
+
+        const { response, data, error } = await GalaxyApi().GET("/api/datasets", {
+            params: {
+                query: {
+                    history_id: selectedHistory.id,
+                    q: ["name-contains"],
+                    qv: [query],
+                    order: `${querySortBy}${sortPrefix}`,
+                    offset: offset,
+                    limit: limit,
+                },
+            },
+        });
+
+        if (error) {
+            errorMessage.value = errorMessageAsString(error);
+            return [];
+        }
+
+        totalItems.value = parseInt(response.headers.get("total_matches") ?? "0");
+
+        items.value = (data as HDASummary[]).map(datasetEntryToRecord);
+
+        formatRows();
+
+        return items.value;
+    } catch (error) {
+        errorMessage.value = errorMessageAsString(error);
+
+        return [];
+    } finally {
+        loading.value = false;
+    }
+}
+
+function onHistoryClick(item: SelectionItem) {
+    if (!item.isLeaf) {
+        selectionDialog.value?.resetFilter();
```

Comment:
This needs to be called using an event handler and not by calling the component function through a ref.

Reply:
The method is being called inside `SelectionDialog`, but ideally, `SelectionDialog` should handle this internally more generically. It might need some refactoring to properly manage such cases without relying on refs.